### PR TITLE
Automation for Reason

### PR DESCRIPTION
**Pull request by bryphe**
Add automated test case for completion with the `basic-reason` template

### CODE REVIEW

**Comment by bryphe** [comment]
Looks like this is _finally_ stabilized! Going to run the test once more to make sure.

Looking forward to when I can move this to be run cross-platform (right now, it only runs on OSX).

**Comment by bryphe** [comment]
2 passes in a row... looks like we're in business

File: build/script/travis-build.sh [hunk 1]
```diff
@@ -27,12 +27,24 @@ npm run pack
 
 echo Using neovim path: "$ONI_NEOVIM_PATH"
 
-npm run test:integration
-
 if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+   echo Installing reason CLI tools...
+   npm install -g https://github.com/reasonml/reason-cli/archive/3.0.4-bin-darwin.tar.gz
```
Comment (jordwalke) [inline build/script/travis-build.sh]: You may want to try the newest `npm install -g reason-cli@3.1.0-darwin`
(Faster download, more updated merlin).

Comment (bryphe) [reply build/script/travis-build.sh]: Excellent! Thanks @jordwalke - testing it out now.

**Pull request merged**